hydroflow/tests/lattice_join.rs

```diff
@@ -0,0 +1,134 @@
+use hydroflow::util::collect_ready;
+use hydroflow_macro::hydroflow_syntax;
+use multiplatform_test::multiplatform_test;
+
+#[multiplatform_test]
+pub fn test_lattice_join_reducing_behavior() {
+    // lattice_fold has this particular behavior where it uses LatticeFrom to convert the first element into the another lattice type and then it folds into that accumulator.
+    // This is to support things like SetUnionSingletonSet -> SetUnionHashSet. It also helps with things like Min, where there is not obvious default value so you want reduce-like behavior.
+    // There's kind of 3 ways you can do this reducing thing, you can do Fold(default_fn, fold_fn), Reduce(reduce_fn), and the third way which is halfway between the two previous approaches, FoldFrom(From_fn, Fold_fn).
+    // The characteristics of the three are, Fold requires a default function, Reduce requires Fn(&mut X, X) -> X, and FoldFrom requires a converter function From_fn: Fn(X) -> A, Fold_fn: Fn(&mut A, X)
```

Comment:
The last two lines should go somewhere for sure. Is this the right place?

Reply:
This text is repeated in the join_fused operator documentation so i will just delete it from here